library-jpa/src/main/java/com/edu/library/model/util/ReadXMLFile.java

```diff
@@ -0,0 +1,180 @@
+
+			for (final Element bookElement : bookListElements) {
+				final Book book = new Book();
+
+				setParameters(bookElement, book);
+				book.setAuthors(addAuthorsParam(bookElement));
+
+				publicationsList.add(book);
+			}
+
+			final List<Element> newsListElements = root.getChildren("Newspaper");
+			new ArrayList<>();
+			for (final Element newsElement : newsListElements) {
+				final Newspaper news = new Newspaper();
+
+				setParameters(newsElement, news);
+				publicationsList.add(news);
+			}
+			return publicationsList;
+		} catch (final Exception e) {
+			e.printStackTrace();
+		}
+		return null;
+
+	}
+
+	/**
+	 * Reads the information of an Author object from the file into a list.
+	 *
+	 * @param publicationElement
+	 *            - the XML element from which the authors will be read
+	 * @return - a list of all the authors for the XML element given by
+	 *         {@code publicationElement}
+	 */
+	private static List<Author> addAuthorsParam(final Element publicationElement) {
+		final List<Element> authorList = publicationElement.getChildren("author");
+		final List<Author> authors = new ArrayList<>();
+		for (final Element element : authorList) {
+			final Author author = new Author();
+
+			author.setUuid(element.getAttributeValue("uuid"));
+			author.setName(element.getChildText("name"));
+			authors.add(author);
+		}
+		return authors;
+	}
+
+	/**
+	 * Reads multiple parameters from the XML element provided by
+	 * {@code publicationElement} and invokes the setter methods of the
+	 * Publication given by {@code publication}
+	 *
+	 * @param publicationElement
+	 *            - the XML element from which the data is read
+	 * @param publication
+	 *            - the Publication type object for which the read attributes
+	 *            will be set
+	 */
+	private static void setParameters(final Element publicationElement, final Publication publication) {
+		publication.setUuid(publicationElement.getAttributeValue("uuid"));
+		publication.setTitle(publicationElement.getChildText("title"));
+		publication.setNrOfCopys(Integer.parseInt(publicationElement.getChildText("nrOfCopies")));
+		publication.setOnStock(Integer.parseInt(publicationElement.getChildText("onStock")));
+
+		final List<Element> publisherList = publicationElement.getChildren("publisher");
+		final List<Publisher> publishers = new ArrayList<>();
+		for (final Element element : publisherList) {
+			final Publisher publisher = new Publisher();
+
+			publisher.setUuid(element.getAttributeValue("uuid"));
+			publisher.setName(element.getChildText("name"));
+			publishers.add(publisher);
+		}
+		publication.setPublisher(publishers.get(0));
+
+		final String date = publicationElement.getChildText("publicationDate");
+
+		final DateFormat df = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH);
+		Date result = null;
+		try {
+			result = df.parse(date);
+		} catch (final ParseException e) {
+			e.printStackTrace();

```

Comment:
itt is csak az e.printStackTrace() van meghivva. Logolni az exception-t, vagy tovabb dobni, vagy egy error uzenetet megjeleniteni stb.